rackcity/utils/errors_utils.py

```diff
@@ -106,11 +110,14 @@ def parse_serializer_errors(errors):
     return " ".join(failure_messages)
 
 
-def parse_validation_error(error):
+def parse_save_validation_error(error, object_name):
     failure_detail = ""
     if isinstance(error, ValidationError):
         for err in error:
             failure_detail += err
+    elif isinstance(error, IntegrityError):
+        for err in error.args:
+            failure_detail += err
     else:
-        failure_detail = GenericFailure.UNKNOWN.value
+        failure_detail = object_name + GenericFailure.ON_SAVE.value
```

Comment:
only used when this is NOT a ValidationError (custom validation errors we have created) or IntegrityError (error thrown from the depths of the DB on save). no idea when that would even happen